db/sql/00_msar.sql

```diff
@@ -974,6 +974,35 @@ SELECT COALESCE(jsonb_agg(priv_cte.p), '[]'::jsonb) FROM priv_cte;
 $$ LANGUAGE SQL STABLE RETURNS NULL ON NULL INPUT;
 
 
+CREATE OR REPLACE FUNCTION msar.get_curr_role_db_priv(db_name text) RETURNS jsonb AS $$/*
+Given a database name, returns a json object with database owner oid and database privileges
+for the role executing the function.
+
+The returned JSON object has the form:
+  {
+    "owner_oid": <int>,
+    "current_role_db_priv" [<str>]
+  }
+*/
+WITH priv_cte AS (
+  SELECT unnest(
+    ARRAY[
+      CASE WHEN has_database_privilege(pgd.oid, 'CREATE') THEN 'CREATE' END,
+      CASE WHEN has_database_privilege(pgd.oid, 'TEMPORARY') THEN 'TEMPORARY' END,
+      CASE WHEN has_database_privilege(pgd.oid, 'CONNECT') THEN 'CONNECT' END
+    ]
+  ) AS p 
+  FROM pg_catalog.pg_database AS pgd WHERE pgd.datname = db_name
+)
+SELECT jsonb_build_object(
+  'owner_oid', pgd.datdba,
+  'current_role_db_priv', COALESCE(jsonb_agg(priv_cte.p), '[]'::jsonb)
+) FROM pg_catalog.pg_database AS pgd, priv_cte
+WHERE pgd.datname = db_name AND priv_cte.p IS NOT NULL
+GROUP BY pgd.datdba;
```

Comment:
I think we can avoid the CTE. I don't see a reason for it to be there since we're querying from the same table.

```sql
SELECT jsonb_build_object(
  'owner_oid',pgd.datdba,
  'current_role_db_priv',
    array_remove(array [
      CASE WHEN has_database_privilege(pgd.oid, 'CREATE') THEN 'CREATE' else null end,
      CASE WHEN has_database_privilege(pgd.oid, 'TEMPORARY') THEN 'TEMPORARY' else null end,
      CASE WHEN has_database_privilege(pgd.oid, 'CONNECT') THEN 'CONNECT' else null end
    ], null)
) FROM pg_catalog.pg_database AS pgd
WHERE pgd.datname = 'mathesar'
GROUP BY pgd.datdba, pgd.oid;
```

This query should work. It also removes `null` from the array of privileges which the existing query retains.

Reply:
Initially, I was trying to implement this without using a CTE, but then I switched to the CTE version for some reason. However, I've now reverted to the non-CTE version.